src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -143,7 +167,8 @@ private static RSACryptoServiceProvider GetCryptoProviderForSha256_Net45(X509Cer
                 // With this flag, a CryptographicException is thrown instead.
                 //
                 csp.Flags |= CspProviderFlags.UseExistingKey;
-                return new RSACryptoServiceProvider(csp);
+                CertificateToRsaCspMap[certificate.Thumbprint] = new RSACryptoServiceProvider(csp);
```

Comment:
Do we know if the RsaCryptoServiceProvider we just created is faulty (e.g. fails on Sign ops)?
If it is faulty, MSAL would throw each time when the RSA object is used.. until the pass-trough cache gets cleared.

Reply:
@GeoK that is a good point, we should only add after a successful usage.